File: cpp/samchon/protocol/master/PRInvokeHistory.cpp
#include <samchon/protocol/master/PRInvokeHistory.hpp>

using namespace std;
using namespace samchon;
using namespace samchon::library;
using namespace samchon::protocol;
using namespace samchon::protocol::master;

PRInvokeHistory::PRInvokeHistory()
	: super()
{
}

PRInvokeHistory::PRInvokeHistory(shared_ptr<Invoke> invoke)
	: super(invoke)
{
	index = invoke->get("piece_index")->getValue<size_t>();
	size = invoke->get("piece_size")->getValue<size_t>();
}

PRInvokeHistory::~PRInvokeHistory()
{
}

File: ts/src/samchon/protocol/master/PRInvokeHistory.ts
/// <reference path="../../API.ts" />

/// <reference path="../InvokeHistory.ts" />

namespace samchon.protocol.master
{
	export class PRInvokeHistory extends InvokeHistory
	{
		private index: number;
		private size: number;

		/* ---------------------------------------------------------
			CONSTRUCTORS
		--------------------------------------------------------- */
		/**
		 * Default Constructor.
		 */
		public constructor();

		/**
		 * Construct from an Invoke message.
		 * 
		 * @param invoke
		 */
		public constructor(invoke: Invoke);

		public constructor(invoke: Invoke = null)
		{
			super(invoke);
			
			if (invoke == null)
			{
				this.index = 0;
				this.size = 0;
			}
			else
			{
				this.index = invoke.get("piece_index").getValue() as number;
				this.size = invoke.get("piece_size").getValue() as number;
			}
		}
		
		public getIndex(): number
		{
			return this.index;
		}

		public getSize(): number
		{
			return this.size;
		}
	}
}